routes/student.mjs

```diff
@@ -36,4 +41,39 @@ router.get("/" ,(req,res)=>{
     res.render('./students/index');
 });
 
+router.post('/login', function(request,response){
+    
+    //names of the input text fields in the views/index.ejs
+    const username = request.body.username;
+    const password = request.body.password;
+    // Query the MySQL database for the student user record
+    const query = 'SELECT userID,password,salt FROM userstudents WHERE sr_code = ?';
+    db.query(query,[username], function(error,result){
+         // If the user is found, return the user's record
+        if (result.length > 0) {
+            //checking of password and salt
+            for(var passCount = 0; passCount < result.length; passCount++){
+                const salt = result[passCount].salt;
+                const passwordHash = password+salt;
+                const dbPassword = result[passCount].password;
+                const has = crypto.createHash('sha256');
+                // Update the hash with the data
+                has.update(passwordHash);
+                // Calculate the hexadecimal hash
+                const hashedSaltAndPass = has.digest('hex')
+                if (dbPassword == hashedSaltAndPass) {
+                    console.log("Login Successful!!!");
+                    response.redirect("dashboard");
+                } else {
+                    //Start of debugging of SHA
+                    console.log("Wrong Password");
+                }
+            }
+        }else{
+            console.log("Wrong email or No emails found")
+            console.log(error);
+        }
```

Comment:
Inverse the conditional statement next time
```Example
database.query(query, [username], (error, result)=>{
if(result.length == 0) response.status(401).send('Login Failed');

//Code continues...
} )



```

nonetheless code looks fine but needs some adjustments